src/cleanfigure.m

```diff
@@ -298,73 +305,300 @@ function pruneOutsideBox(meta, handle)
+    % simplification.
+
+    % Points per pixel retained before the line simplification. In other
+    % words, how much you can zoom-in before losing resolution.
+    ZOOM_MULTIPLIER = 4;
+
+    % Do not simpify
+    if any(isinf(targetResolution))
+        return
+    end
 
-  % Extract the data from the current line handle.
-  xData = get(handle, 'XData');
-  yData = get(handle, 'YData');
-  zData = get(handle, 'ZData');
-  if ~isempty(zData)
-    % Don't do funny stuff when zData is present.
-    return;
-  end
+    % Extract the data from the current line handle.
+    xData = get(handle, 'XData');
+    yData = get(handle, 'YData');
+    zData = get(handle, 'ZData');
 
-  data = [xData(:), yData(:)];
+    if ~isempty(zData)
+        % TODO: 3d simplificattion of frontal 2d projection
+        return;
+    end
+    if isempty(xData) || isempty(yData)
+        return;
+    end
+    if numel(xData) <= 2
+        return;
+    end
 
-  if isempty(data)
-      return;
-  end
+    % Get info about log scaling
+    isXlog = strcmp(get(meta.gca, 'XScale'), 'log');
+    vxData = xData;
+    if isXlog
+        vxData = log10(xData);
+    end
 
-  % Generate a mask which is true for the first point, and all
-  % subsequent points which have a greater norm2-distance from
-  % the previous point than 'threshold'.
-  n = size(data, 1);
-
-  % Get info about log scaling.
-  isXlog = strcmp(get(meta.gca, 'XScale'), 'log');
-  isYlog = strcmp(get(meta.gca, 'YScale'), 'log');
-
-  mask = false(n, 1);
-
-  XRef = data(1,:);
-  mask(1) = true;
-  for kk = 2:n
-      % Compute the visible distance of those points,
-      % incorporating possible log-scaling of the axes.
-      visDiff = XRef - data(kk,:);
-      if isXlog
-          % visDiff(1) = log10(XRef(1)) - log10(data(kk,1));
-          visDiff(1) = log10(visDiff(1));
-      end
-      if isYlog
-          visDiff(2) = log10(visDiff(2));
-      end
-      % Check if it's larger than the threshold and
-      % update the reference point in that case.
-      if norm(visDiff) > minimumPointsDistance
-          XRef = data(kk,:);
-          mask(kk) = true;
-      end
-  end
-  mask(end) = true;
+    isYlog = strcmp(get(meta.gca, 'YScale'), 'log');
+    vyData = yData;
+    if isYlog
+        vyData = log10(yData);
+    end
 
-  % Make sure to keep NaNs.
-  mask = mask | any(isnan(data)')';
+    % Automatically guess a tol based on the area of the figure and
+    % the area and resolution of the output
+    a = axis(meta.gca);
+    if ~isXlog
+        xrange = (a(2)-a(1));
+    else
+        xrange = (log10(a(2))-log10(a(1)));
+    end
+    if ~isYlog
+        yrange = (a(4)-a(3));
+    else
+        yrange = (log10(a(4))-log10(a(3)));
+    end
+    tol = xrange*yrange/(ZOOM_MULTIPLIER^2*prod(targetResolution));
+    nPixelsX = targetResolution(1)*sqrt(targetResolution(3));
+    nPixelsY = targetResolution(2)*sqrt(targetResolution(3));
```

Comment:
I realise I made a mistake here. It's generally accepted to use resolution in pixels/points per (linear) length, not per area. We should change `sqrt(targetResolution(3))` to `targetResolution(3)` and `prod(targetResolution)` to `prod(targetResolution)*targetResolution(3)` to be consistent with standard use.

Alternatively, we go with as I suggested earlier and make `targetResolution` as simply the width and height of the target figure in pixels. This seems like a simpler interface.